File: app/src/main/java/com/allerates/appforyou/activities/AccountActivity.java
package com.allerates.appforyou.activities;

import android.content.Intent;
import android.os.Bundle;
import android.support.annotation.NonNull;
import android.support.v7.app.AppCompatActivity;
import android.view.View;
import android.widget.Button;

import com.allerates.appforyou.R;
import com.google.firebase.auth.FirebaseAuth;

public class AccountActivity extends AppCompatActivity {
    private Button mLogOut;
    private FirebaseAuth mAuth;
    private FirebaseAuth.AuthStateListener mAuthStateListener;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_account);
        mLogOut = (Button) findViewById(R.id.logOutBtn);
        mAuth = FirebaseAuth.getInstance();
        mAuthStateListener = new FirebaseAuth.AuthStateListener() {
            @Override
            public void onAuthStateChanged(@NonNull FirebaseAuth firebaseAuth) {
            if(firebaseAuth.getCurrentUser() == null){
                startActivity(new Intent(AccountActivity.this, MainActivity.class));
            }
            }
        };

        mLogOut.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                mAuth.signOut();
            }
        });
    }

    @Override
    protected void onStart() {
        super.onStart();

        mAuth.addAuthStateListener(mAuthStateListener);
    }
}

File: AppCurrencyAndroid/app/src/main/java/com/allerates/appforyou/activities/AccountActivity.kt
package com.allerates.appforyou.activities


import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import android.view.View
import com.allerates.appforyou.helpers.startActivity
import com.allerates.appforyou.R
import com.google.firebase.auth.FirebaseAuth


class AccountActivity : AppCompatActivity() {

    private var mAuth: FirebaseAuth? = null
    private var mAuthStateListener: FirebaseAuth.AuthStateListener? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_account)

        mAuth = FirebaseAuth.getInstance()
        mAuthStateListener = FirebaseAuth.AuthStateListener { firebaseAuth ->
            if (firebaseAuth.currentUser == null)
                startActivity(MainActivity::class.java)
        }
    }

    fun logOutClick(v: View) = mAuth!!.signOut()

    override fun onStart() {
        super.onStart()
        mAuth!!.addAuthStateListener(mAuthStateListener!!)
    }
}
